src/utils/mtev_skiplist.c

```diff
@@ -191,14 +191,20 @@ static int mtev_skiplisti_find_compare(mtev_skiplist *sl,
       m=m->next; /* m->next is the match */
       while(m->down) m=m->down; /* proceed to the bottom-most */
       if(ret) *ret = m;
-      if(prev) *prev = m->prev;
+      /* We have to be careful when setting *prev: the first column is a
+       * place-holder element with NULL data that we don't want to expose
+       * to clients. So check that m->prev isn't the first column before
+       * filling in *prev. */
+      if(prev && m->prev && m->prev->prev)
+        *prev = m->prev;
```

Comment:
We don't want to skip sething it, instead we want to set it to null.

If (prev)
 *prev = (m->prev && !m->prev->prev) ? NULL : m->prev;

I think